src/filters/ColorMatrixFilters.ts

```diff
@@ -1,38 +1,27 @@
-import { ColorMatrix } from './ColorMatrix';
+import { ColorMatrix, colorMatrixDefaultValues } from './ColorMatrix';
 import { classRegistry } from '../ClassRegistry';
 
 export function createColorMatrixFilter(key: string, matrix: number[]) {
-  return class GeneratedColorMatrix extends ColorMatrix {
-    /**
-     * Filter type
-     * @param {String} type
-     * @default
-     */
-    type = key;
-
-    /**
-     * Colormatrix for the effect
-     * array of 20 floats. Numbers in positions 4, 9, 14, 19 loose meaning
-     * outside the -1, 1 range.
-     * @param {Array} matrix array of 20 numbers.
-     * @default
-     */
-    matrix = matrix;
-
-    /**
-     * Lock the matrix export for this kind of static, parameter less filters.
-     */
-    mainParameter = undefined;
+  const newClass = class extends ColorMatrix {
+    get type() {
+      return key;
+    }
 
-    /**
-     * Lock the colormatrix on the color part, skipping alpha
-     */
-    colorsOnly = true;
+    static defaults = {
+      ...colorMatrixDefaultValues,
+      /**
+       * Lock the matrix export for this kind of static, parameter less filters.
+       */
+      mainParameter: undefined,
+      matrix,
+    };
 
     static async fromObject(object: any) {
-      return new GeneratedColorMatrix(object);
+      return new this(object);
     }
   };
+  classRegistry.setClass(newClass, key);
+  return newClass;
 }
```

Comment:
I would make this not a mixin but a subclass
See https://github.com/fabricjs/fabric.js/blob/08e9431b6c19f763216f9061d7cec25774c58961/src/filters/ColorMatrixFilters.ts

Reply:
mixins suck in TS

Reply:
but this is not a mixin. this is a subclass generated with a function.

Reply:
I meant dynamic classes, sorry